spec.bs

```diff
@@ -829,6 +835,19 @@ An interest group is a [=struct=] with the following items:
   which includes some slots that can be filled in with specific "products".
 : <dfn>joining origin</dfn>
 :: An [=url/origin=].The top level page origin from where the interest group was joined.
+: <dfn>join counts</dfn>
+:: A [=list=] containing [=tuple=]s of the day and per day join count. The day

```

Comment:
would it be simpler to just have a list of [=date=]?  then finding the joinCount is just finding the count of list entries less than 30 days ago?

Reply:
It would be simpler, but significantly less compact, storage-wise. We were seeing some publishers calling joinAdinterestGroup() on *every page load*, which was significantly increasing interest group storage size and causing counting the list of entries to become a performance bottleneck. A per-day count hopefully balances utility and expressiveness.

Reply:
Is "how the count/list is internally represented" an implementation detail we don't need to get into?  My comment was optimizing for simplest way to explain it.  Anyhow I don't feel strongly about it.  Feel free to resolve this thread without changes if necessary.

Reply:
The issue is that there is a difference in how the counts are expired. If we include a full date including time then expiration would be spread throughout a day. But with the per-day counts we drop a full day's worth of counts all at once.